### PR TITLE
Materials pagination

### PR DESCRIPTION
**Pull request by bandvov**
this pull-request should be merged before `materials-permissions` pull-request

### CODE REVIEW

File: modules/comment/comment.service.js [hunk 1]
```diff
@@ -50,26 +50,26 @@ class CommentsService {
     const product = await Product.findById(id);
     const { userRates } = product;
     let { rateCount } = product;
-    const { rate } = userRates.find(rate => String(rate.user) === String(user._id)) || {};
+    const { rate } =      userRates.find(rate => String(rate.user) === String(user._id)) || {};
```
Comment (TorskyM) [inline modules/comment/comment.service.js:53]: Remove unnecessary space

File: modules/comment/comment.service.js [hunk 2]
```diff
@@ -50,26 +50,26 @@ class CommentsService {
...
 
     const newUserRates = rate
-        ? userRates.map(item => (String(item.user) === String(user._id)
-          ? { user: item.user, rate: data.rate }
-          : item))
-        : [...userRates, { ...data, user: user._id }]
+      ? userRates.map(item => (String(item.user) === String(user._id)
+        ? { user: item.user, rate: data.rate }
+        : item))
+      : [...userRates, { ...data, user: user._id }];
```
Comment (TorskyM) [inline modules/comment/comment.service.js:63]: It's better to replace this nested ternary operator with `if` statement so it'll become more readable this way

**Review by TorskyM — APPROVED** [review]

**Review by TarasKun — APPROVED** [review]

**Comment by Alan18081** [comment]
Please, resolve conflicts and add tests for materials pagination in `material.query.test.js`

File: __tests__/materials/material.query.test.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                count
+              }
+            }
+          `,
+        })
+        .catch(e => e);
+
+      const { items } = res.data.getAllMaterials;
+      expect(items.length).toEqual(6);
+      expect(items).not.toBeNull();
```
Comment (Alan18081) [inline __tests__/materials/material.query.test.js:223]: Please, also, check `count` here

Comment (bandvov) [reply __tests__/materials/material.query.test.js:225]: fixed

**Pull request merged**